From 57c0539d99fed2252364b431de004f9110ac3d3f Mon Sep 17 00:00:00 2001
From: Vladimir Oltean <vladimir.oltean@nxp.com>
Date: Fri, 6 Dec 2019 13:59:05 +0200
Subject: [PATCH] arm64: dts: ls1028a-rdb: Enable SGMII AN for the QSGMII
 switch ports

This enables monitoring of link status and AN. It should also physically
enable SGMII AN with the VSC8514 PHY, but in practice that is still
hardcoded as "on" in the PHY driver, at the moment. So since Felix
actually disables SGMII AN when this DT property is absent, this would
result in an in-band AN mismatch between the MAC and the PHY. So this
property is required for the moment for this MAC/PHY combination.

Signed-off-by: Vladimir Oltean <vladimir.oltean@nxp.com>
---
 arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts | 4 ++++
 1 file changed, 4 insertions(+)

--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
@@ -236,21 +236,25 @@
 &switch_port0 {
 	phy-handle = <&qsgmii_phy1>;
 	phy-connection-type = "qsgmii";
+	managed = "in-band-status";
 };
 
 &switch_port1 {
 	phy-handle = <&qsgmii_phy2>;
 	phy-connection-type = "qsgmii";
+	managed = "in-band-status";
 };
 
 &switch_port2 {
 	phy-handle = <&qsgmii_phy3>;
 	phy-connection-type = "qsgmii";
+	managed = "in-band-status";
 };
 
 &switch_port3 {
 	phy-handle = <&qsgmii_phy4>;
 	phy-connection-type = "qsgmii";
+	managed = "in-band-status";
 };
 
 &sai4 {
